internal/connect/system.go

```diff
@@ -124,3 +135,37 @@ func UpdateCertificates() error {
 	// reload CA certs in Go
 	return ReloadCertPool()
 }
+
+// ReadOnlyFilesystem returns an error if the given root path contains a
+// read-only mount point or if the system should actually be managed through
+// `transactional-update`. Otherwise it just returns nil. Note that if the given
+// root path is empty, then "/" is assumed.
+func ReadOnlyFilesystem(root string) error {
+	path := root
+	if path == "" {
+		path = "/"
+	}
+	statfs := &syscall.Statfs_t{}
+	if err := syscall.Statfs(path, statfs); err != nil {
+		return fmt.Errorf("Checking whether %v is mounted read-only failed: %v", path, err)
+	}
+
+	// Just like zypper, we will assume that a BTRFS file system with the
+	// `transactional-update` binary installed is a transactional server.
+	_, err := os.Stat("/usr/sbin/transactional-update")
+	if statfs.Type == BTRFS_SUPER_MAGIC && err == nil {
+		// The user did not use the 'root' flag from the CLI: this is not
+		// `transactional-update` calling SUSEConnect but rather a user
+		// directly. This is not allowed.
+		if root == "" {
```

Comment:
```suggestion
		if path == "/" {
```
I would change it to this, to make the behavior more clear (at least for me).

Reply:
I chose `root` here because of the implicit meaning of it: `path` is what we compute, `root` is what the user provided. Hence, the if reads as "if the user did not provide anything".

As for the `--root /`, I'd say we follow the general logic: the user provided the `--root` flag (they know what they are doing, *hopefully*) and hence, if it's read only, it will be catched on the if down below. The general flow of the card was: if the user does not provide the root flag, then it better not be on a transactional system; otherwise, if they provide the flag, it's either `transactional-update` or someone doing something *special*, which we should only forbid if it's read only.

Reply:
I see your point. Fair enough!